src/Http/Curators/CreateNewsArticleRequestHandler.php

```diff
@@ -48,9 +51,15 @@ public function handle(ServerRequestInterface $request): ResponseInterface
 
         $this->newsArticleRepository->create($newsArticle);
 
-        return new JsonResponse(
-            ['id' => $newsArticle->getId()->toString()],
-            StatusCodeInterface::STATUS_CREATED
-        );
+        $headers = new Headers($request);
+        $responseContentType = $headers->determineResponseContentType(['application/ld+json', 'application/json']);
+        $withJsonLd = $responseContentType === 'application/ld+json';
+
+        $normalized = (new NewsArticleNormalizer())->withJsonLd($withJsonLd)->normalize($newsArticle);
```

Comment:
Maybe the construction of the `NewsArticleNormalizer` can move to a very simple factory class so we avoid some duplication?

Reply:
Yes, I was also thinking how I could reduce the duplication yesterday but didn't find a good way because we also need the `$withJsonLd` boolean to determine the right `Response` class to return (`JsonResponse` or `JsonLdResponse`). But I think I found a good solution, I'll try it out.